### ISSUE TITLE
conf->hostname might be uninitialized in be_jwt_init()

### ISSUE THREAD

**Issue by rmilenko** [jpmens/mosquitto-auth-plug#383]
What happens when there's no http_hostname parameter in mosquitto.conf file for JWT backend?

```
be_jwt_init()

        conf = (struct jwt_backend *)malloc(sizeof(struct jwt_backend));
        conf->ip = ip;
        conf->port = p_stab("http_port") == NULL ? 80 : atoi(p_stab("http_port"));
        if (p_stab("http_hostname") != NULL) {
                conf->hostheader = (char *)malloc(128);
                conf->hostname = p_stab("http_hostname");
                sprintf(conf->hostheader, "Host: %s", p_stab("http_hostname"));
        } else {
                conf->hostheader = NULL;
        }

```

I believe something like 

`conf->hostname = NULL;`

should be added in 'else' statement along with 'conf->hostheader = NULL;'

Otherwise, http_post() will assume 'conf->hostname' IS NOT NULL and will try to use it instead of 'conf->ip', in snprintf().

Workaround is to set both http_ip and http_hostname to same value in mosquitto.conf for JWT backend.
Of course, this is only the case when TLS is not used.

**Comment by rmilenko** [comment]
I believe this issue can be closed, now that commit [4f67cfbe59d619adbce600430bb6aa6f775530d7](https://github.com/jpmens/mosquitto-auth-plug/commit/4f67cfbe59d619adbce600430bb6aa6f775530d7) exists, which implements hostheader and hostname initialization.

Thank you 👍

**Comment by jpmens** [comment]
Thank you for tracking this.
